src/larecs/comptime_optional.mojo

```diff
@@ -109,6 +109,23 @@ struct ComptimeOptional[
         )
         return UnsafePointer(ptr)[]
 
+    @always_inline
+    fn or_else(self, value: ElementType) -> ElementType:
```

Comment:
Just as a note, you may want to take a look how Rust does this.

In Rust, this would be `or`, while `or_else` takes a callback. So the argument to `or` is always evaluated. Contrary, `or_else` evaluates the argument only in the else-case, which makes sense when constructing the argument has an associated cost.